File: Leetcode/array/RemoveDuplicatesFromSortedArrayII/src/python/RemoveDuplicatesFromSortedArrayII.py
class Solution(object):
    def removeDuplicates(self, nums):
        if not nums:
            return 0
        
        last,i,same=0,1,False
        while i<len(nums):
            if nums[last]!=nums[i] or not same:
                same=nums[last]==nums[i]
                last+=1;
                nums[last]=nums[i]
            i+=1
        return last+1
if __name__=="__main__":
    print Solution().removeDuplicates([1,1,1,2,2,3])

File: Leetcode/array/RemoveDuplicatesFromSortedArrayII/src/cpp/RemoveDuplicatesFromSortedArrayII.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size()==0)
            return nums.size();
        int i=0,last=2;
        for (int index=1,temp=0;index<nums.size();index++){
            if (nums[i]!=nums[index]){
                nums[++i]=nums[index];
                temp=0;
            }
            else{
                temp++;
                if (temp<last)
                    nums[++i]=nums[index];
            }
        }
        return i+1;
        
    }
};